streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -480,7 +495,8 @@ void commit(final boolean startNewTransaction) {
         for (final Map.Entry<TopicPartition, Long> entry : consumedOffsets.entrySet()) {
             final TopicPartition partition = entry.getKey();
             final long offset = entry.getValue() + 1;
-            consumedOffsetsAndMetadata.put(partition, new OffsetAndMetadata(offset));
+            consumedOffsetsAndMetadata.put(partition, new OffsetAndMetadata(offset,
+                                                                            ((Long) getPartitionTime(partition)).toString()));
```

Comment:
To be future prove, we should encode a version number as prefix in case we ever what to change this metadata. What about `<version>:<partitionTime>` with version number "1" ?

Also, line is too long. Move both parameters to their own lines.

Reply:
@guozhangwang @bbejeck @vvcephei WDYT about this?